larq/callbacks.py

```diff
@@ -52,20 +64,32 @@ def set_model(self, model: keras.models.Model):
                 f'Optimizer must have a "{self.hyperparameter}" attribute.'
             )
 
-    def on_epoch_begin(self, epoch: int, logs: Optional[Dict] = None):
+    def set_hyperparameter(self, t: int):
         hp = getattr(self.optimizer, self.hyperparameter)
         try:  # new API
             hyperparameter_val = tf.keras.backend.get_value(hp)
-            hyperparameter_val = self.schedule(epoch, hyperparameter_val)
+            hyperparameter_val = self.schedule(t, hyperparameter_val)
         except TypeError:  # Support for old API for backward compatibility
-            hyperparameter_val = self.schedule(epoch)
+            hyperparameter_val = self.schedule(t)
 
         tf.keras.backend.set_value(hp, hyperparameter_val)
+        return hp
 
-        if self.verbose > 0:
-            print(
-                f"Epoch {epoch + 1}: {self.hyperparameter} changing to {tf.keras.backend.get_value(hp)}."
-            )
+    def on_epoch_begin(self, epoch: int, logs: Optional[Dict] = None):
+        if self.unit == "epoch":
+            hp = self.set_hyperparameter(epoch)
+
+            if self.verbose > 0:
+                print(
+                    f"Epoch {epoch + 1}: {self.hyperparameter} changing "
+                    + f"to {tf.keras.backend.get_value(hp)}."
+                )
```

Comment:
I don't think we're splitting strings for the 88-char limit, so you can keep these as one line.

Reply:
Also (maybe check this) I think epochs are reported 0-indexed everywhere, so I wouldn't do `epoch + 1` here.

Reply:
This code was already in the hyperparameter before, so I assume this is okay? I splitted the string just because I think it looks uglier if I don't

Reply:
We don't split strings or f-strings anywhere else in the codebase as far as I can tell, so I think it's better to be consistent and also keep this string as one line.

Reply:
I like splitting strings as well - Zookeeper splits all the strings - not that it really matters too much. But I believe if you do choose to split them you can get rid of the `+`.